### PR TITLE
Add Color#withOpacity

### PR DESCRIPTION
**Pull request by abarth**
Fixes https://github.com/flutter/flutter/issues/2277

### CODE REVIEW

**Comment by abarth** [comment]
@HansMuller

File: sky/engine/core/painting/Color.dart [hunk 1]
```diff
@@ -49,22 +49,32 @@ class Color {
...
+  /// replaced with a (which ranges from 0 to 255).
   Color withAlpha(int a) {
     return new Color.fromARGB(a, red, green, blue);
   }
 
-  /// Returns a new color that matches this color with the red channel replaced with r.
+  /// Returns a new color that matches this color with the alpha channel
+  /// replaced with the given opacity (which ranges from 0.0 to 1.0).
+  Color withOpacity(double opacity) {
+    return withAlpha((255.0 * myOpacity).round());
```
Comment (apwilson) [inline sky/engine/core/painting/Color.dart]: where'd myOpacity come from?

**Comment by Hixie** [comment]
lgtm modulo what @apwilson noticed

File: sky/engine/core/painting/Color.dart (hunk 1 above)
Comment (abarth) [inline sky/engine/core/painting/Color.dart]: Oh noes!

**Comment by apwilson** [comment]
Also, should you protect against opacity < 0.0 or > 1.0?

**Comment by abarth** [comment]
> Also, should you protect against opacity < 0.0 or > 1.0?

Sure.

**Comment by apwilson** [comment]
Maybe fromARGB should do that protection since everyone calls it.

**Comment by abarth** [comment]
> Maybe fromARGB should do that protection since everyone calls it.

It's protected in the sense that you get a legal value, but we can't actually add an assert because we want the constructor to be const.  We've requested a change to the Dart language to be able to have asserts in const constructors but that's currently not possible.

**Pull request merged**